### ISSUE TITLE
avoid redirects in the RPC API

### ISSUE THREAD

**Issue by potuz** [prysmaticlabs/prysm#9823]
I am not sure if reporting this as a bug, feature request or security issue. Currently when querying an endpoint with something like `curl -s http://localhost:3500/eth/v1/node/syncing`, the request is being redirected to `http://localhost:3500/internal/eth/v1/node/syncing`. The problem is that these two requests can be made by different users of the system. Namely the first one is made by the user requesting it, the second one is made by the user running the beacon-chain binary. 

This presents a security issue since someone that runs the beacon-chain as a random user would have to allow outgoing packets to `localhost:3500` for every user instead of some fixed sandboxed one. 

To be explicit, in my system I do not allow outgoing connections to localhost, but I do open outgoing connections to localhost:3500 for one single user that is allowed to query the beacon-chain node. This user is not the same user that runs the beacon which is chosen at random by systemd at launch.

**Comment by rkapka** [comment]
Hi @potuz

Unfortunately this is very unlikely to happen. We use the [grpc-gateway library](https://github.com/grpc-ecosystem/grpc-gateway) to translate between gRPC and JSON over HTTP. This library is not flexible enough to build an API which conforms to the official spec. This is why we constructed a middleware proxy that sits between the HTTP client and grpc-gateway.

In order to remove redirects we would need to do one of the following:
- give up on gRPC (not going to happen)
- write our own "grpc-gateway" (not going to happen)
- modify our grpc-gateway fork so that it provides all functionality of the current middleware (very unlikely)

To summarize, I don't think we currently have any options to remove the proxying.
